main.tf

```diff
@@ -3,3 +3,34 @@
 # And it continues with some lowercase information about the module
 # We might add more than one line for additional information
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+resource "google_logging_project_sink" "sink" {
+  count = var.module_enabled ? 1 : 0
+
+  name        = var.name
+  destination = var.destination
+
+  filter                 = var.filter
+  description            = var.description
+  disabled               = var.disabled
+  project                = var.project
+  unique_writer_identity = var.unique_writer_identity
+
+  # TODO: is this correct?
+  bigquery_options {
+    use_partitioned_tables = var.use_partitioned_tables
+  }
```

Comment:
the block should only exist `var.use_partitioned_tables != null` so making it a dynamic bloack that only renders if the variable is set to something other than null